### PR TITLE
Add typing hints

### PR DESCRIPTION
**Pull request by SchoolGuy**
This PR does two things:

- Add typing hints and docstrings to some of the functionality of this repository
- Fix typos in the documentation

### CODE REVIEW

File: osclib/pkglistgen_comments.py [hunk 1]
```diff
@@ -145,7 +184,13 @@ def is_approved(self, comment, comments):
                     return c['who']
         return None
 
-    def parse_title(self, line):
+    def parse_title(self, line: str) -> Optional[Dict[str, Union[str, List[str]]]]:
```
Comment (coolo) [inline osclib/pkglistgen_comments.py:187]: If you want this typed, you should make the commands a class. Shouldn't be a lot of lines

File: osclib/pkglistgen_comments.py [hunk 2]
```diff
@@ -271,7 +361,9 @@ def check_staging_accept(self, project, target):
         sections = self.parse_sections(comment['comment'])
         with tempfile.TemporaryDirectory() as tmpdirname:
             checkout_package(self.apiurl, target, '000package-groups', expand_link=True, outdir=tmpdirname)
+            # Now should go to Attribute OSRT:summary-staging in the package or project
```
Comment (coolo) [inline osclib/pkglistgen_comments.py:364]: Not sure this is true anymore

**Review by coolo — CHANGES_REQUESTED** [review]
Please either avoid the complex types or make them proper class

File: osclib/pkglistgen_comments.py (hunk 2 above)
Comment (SchoolGuy) [reply osclib/pkglistgen_comments.py:364]: Yes I will drop this. This was a WIP comment that slipped into the commit.

File: osclib/pkglistgen_comments.py [hunk 3]
```diff
@@ -226,59 +265,59 @@ def parse_sections(self, comment: str) -> List[Dict[str, Union[str, List[str]]]]
...
         Performs a transformation of the parsed comment to the summary file structure.
 
         :param content: The dict that contains the summary file content.
         :param section: The section that contains a move operation of a package
         """
-        for pkg in section['pkgs']:
+        for pkg in section.pkgs:
             pkg_content = content[pkg]
-            for group in section['from']:
+            for group in section.pkgs:
```
Comment (coolo) [inline osclib/pkglistgen_comments.py:282]: this looks wrong

Comment (SchoolGuy) [reply osclib/pkglistgen_comments.py]: Fixed.

File: osclib/pkglistgen_comments.py [hunk 4]
```diff
@@ -145,19 +214,41 @@ def is_approved(self, comment, comments):
                     return c['who']
         return None
 
-    def parse_title(self, line):
+    def parse_title(self, line: str) -> Optional[PkglistSection]:
+        """
+        Parses the header of a section from a single line of a comment that has been passed.
+
+        :param line: The line that should be checked.
+        :returns: None or a dict with the following structure: {"cmd": "<add|move|remove", "<to|from>": str, "pkgs": []}
```
Comment (coolo) [inline osclib/pkglistgen_comments.py:222]: Did I mention why I dislike docstrings on the base of them getting old very quickly and as such never be trustworthy? What an irony you're demonstrating it live ;)

Comment (SchoolGuy) [reply osclib/pkglistgen_comments.py:222]: Oh damn. Yes indeed you are correct. It seems I have to do another version of this PR. I will remove it as desired from you.

**Review by coolo — APPROVED** [review]

**Pull request merged**